test/integ/deferred.cpp

```diff
@@ -67,7 +67,8 @@ class deferred_resource : public http_resource
     public:
         const shared_ptr<http_response> render_GET(const http_request& req)
         {
-            return shared_ptr<deferred_response>(new deferred_response(test_callback, "cycle callback response"));
+            return shared_ptr<deferred_response>(new deferred_response(test_callback, nullptr, nullptr,
```

Comment:
We should have a separate resource and a separate that is used to verify that the new functionality is working fine.